x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -25,24 +35,42 @@
 import org.elasticsearch.snapshots.SnapshotId;
 import org.elasticsearch.snapshots.SnapshotInfo;
 import org.elasticsearch.snapshots.SnapshotShardFailure;
+import org.elasticsearch.snapshots.SnapshotState;
+import org.elasticsearch.xpack.ccr.Ccr;
+import org.elasticsearch.xpack.ccr.CcrLicenseChecker;
 
 import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
 
 /**
  * This repository relies on a remote cluster for Ccr restores. It is read-only so it can only be used to
  * restore shards/indexes that exist on the remote cluster.
  */
 public class CcrRepository extends AbstractLifecycleComponent implements Repository {
 
+    private static final String SNAPSHOT_UUID = "_latest_";
     public static final String TYPE = "_ccr_";
     public static final String NAME_PREFIX = "_ccr_";
 
     private final RepositoryMetaData metadata;
-
-    public CcrRepository(RepositoryMetaData metadata, Settings settings) {
+    private final String remoteClusterAlias;
+    private final Client client;
+    private final CcrLicenseChecker ccrLicenseChecker;
+    // The snapshot will be compatible because remote cluster connections must be compatible with our cluster
+    // version. So the snapshot version is not too important.
+    // TODO: Evaluate where we eventually want to pull the version from
+    private final Version version = Version.CURRENT;
```

Comment:
the version here matters. As an initial approximation, we can take the maximum version of data nodes in the leader cluster.

Reply:
```
    private void validateSnapshotRestorable(final String repository, final SnapshotInfo snapshotInfo) {
        if (!snapshotInfo.state().restorable()) {
            throw new SnapshotRestoreException(new Snapshot(repository, snapshotInfo.snapshotId()),
                                               "unsupported snapshot state [" + snapshotInfo.state() + "]");
        }
        if (Version.CURRENT.before(snapshotInfo.version())) {
            throw new SnapshotRestoreException(new Snapshot(repository, snapshotInfo.snapshotId()),
                                               "the snapshot was created with Elasticsearch version [" + snapshotInfo.version() +
                                                   "] which is higher than the version of this node [" + Version.CURRENT + "]");
        }
    }
```

Here is the validation performed based on this version. So if the leader cluster is a newer version than the follower cluster, it will be marked as incompatible. So I will take the approach of the maximum version from the data nodes in the leader cluster if you are happy with that.

Reply:
>  So I will take the approach of the maximum version from the data nodes in the leader cluster if you are happy with that.

The validation logic you mentioned above (`validateSnapshotRestorable`) will not guarantee us that there isn't a data node in the follower cluster that is older than the master that runs the validation logic there, but it's at least better than no validation at all. I think it's the best check we can do for now. Can you add an item to the meta-issue that we need to think a bit more about BWC constraints?

Reply:
I will add a note.